src/libs/actions/IOU.ts

```diff
@@ -3691,10 +3691,75 @@ function sendMoneyWithWallet(report: OnyxTypes.Report, amount: number, currency:
     Report.notifyNewAction(params.chatReportID, managerID);
 }
 
+function shouldShowPayButton(iouReport: OnyxEntry<OnyxTypes.Report> | EmptyObject, chatReport: OnyxEntry<OnyxTypes.Report> | EmptyObject, policy: OnyxEntry<OnyxTypes.Policy> | EmptyObject) {
```

Comment:
Loooove these helper methods since they removed duplication and made code easier to follow ❤️

Reply:
Is this really determining if we need to show the button or is it really determining `canIOUBePaid`?

Reply:
Especially [here](https://github.com/Expensify/App/pull/35928/files?w=1#diff-9c07b44e092fd3d975806649be46d76b75a377c16e32ed86597b630a84c290d0R3753) is weird to call a method called shouldShowPayButton